main/accounts/index.ts

```diff
@@ -174,10 +174,11 @@ export class Accounts extends EventEmitter {
       if (!action?.update) return
 
       action.update(request, data)
-      currentAccount.update()
-    } else if (request.type === 'signErc20Permit') {
-      store.updateTypedDataRequest(currentAccount.id, reqId, data)
     }
+    if (request.type === 'signErc20Permit') {
+      Object.assign(currentAccount.requests[reqId], data)
```

Comment:
I don't think we should use `Object.assign` if we can avoid it as its not actually making the assignment you think it is. the output for this call would be something like `RequestType & data` as TS will infer that you want to output a type which is the union of the two if they're not compatible, which is not what we want. can we assign the value directly instead?